src/pages/ReplaceServer/PrepareController.js

```diff
@@ -0,0 +1,180 @@
+import { translate } from '../../localization/localize.js';
+import { STATUS } from '../../utils/constants.js';
+import BaseUpdateWizardPage from '../BaseUpdateWizardPage.js';
+import { PlaybookProgress } from '../../components/PlaybookProcess.js';
+import { ErrorBanner } from '../../components/Messages.js';
+import { fetchJson, postJson, deleteJson } from '../../utils/RestUtils.js';
+
+class PrepareController extends BaseUpdateWizardPage {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      overallStatus: STATUS.UNKNOWN, // overall status of entire playbook and commit
+      invalidMsg: '',
+    };
+  }
+
+  setNextButtonDisabled = () => this.state.overallStatus != STATUS.COMPLETE;
+
+  updatePageStatus = (status) => {
+    this.setState({overallStatus: status});
+    if (status === STATUS.FAILED) {
+      this.setState({invalidMsg: translate('server.replace.prepare.failure')});
+    }
+  }
+
+  renderPlaybookProgress () {
+    const steps = [
+      {
+        label: translate('deploy.progress.commit'),
+        playbooks: ['commit']
+      },
+      {
+        label: translate('deploy.progress.config-processor-run'),
+        playbooks: ['config-processor-run.yml']
+      },
+      {
+        label: translate('deploy.progress.ready-deployment'),
+        playbooks: ['ready-deployment.yml']
+      },
+      {
+        label: translate('server.deploy.progress.rm-cobbler'),
+        playbooks: ['rm-cobbler']
+      },
+      {
+        label: translate('server.deploy.progress.rm-known-host'),
+        playbooks: ['known-hosts']
+      },
+      {
+        label: translate('server.deploy.progress.cobbler-deploy'),
+        playbooks: ['cobbler-deploy.yml']
+        // TODO:
+        //    cobbler-deploy *might* prompt for a password, although the logs on the QE system
+        //       ardana@10.84.81.17 indicate that none was supplied and it succesfully ran. Hmmmm
+      },
+    ];
+
+    const playbooks = [
+      {
+        name: 'commit',
+        action: ((logger) => {
+          const commitMessage = {'message': 'Committed via Ardana Installer'};
+          return postJson('/api/v1/clm/model/commit', commitMessage)
+            .then((response) => {
+              logger('Model committed\n');
+            })
+            .catch((error) => {
+              const message = translate('update.commit.failure', error.toString());
+              logger(message+'\n');
+              throw new Error(message);
+            });
+        }),
+      },
+      {
+        name: 'config-processor-run',
+      },
+      {
+        name: 'ready-deployment',
+      },
+      {
+        name: 'rm-cobbler',
+        action: ((logger) => {
+          return fetchJson('/api/v1/clm/cobbler/servers')
+            .then((response) => {
+              const cobbler_server = response.find((e) =>
+                e.ip === this.props.operationProps.server.ip ||
+                e.name === this.props.operationProps.server.name);
```

Comment:
this.props.operationProps.server.name should be this.props.operationProps.server.id which is the model server id matching to the cobbler name

Reply:
Good catch. Will fix